### ISSUE TITLE
InlineCompletion partialAccept not firing correctly with Suggest dropdown

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#32432]
TLDR; Accepting Suggest Item (CompletionItem) fires partialAccept, but next line/next token acceptance fails to trigger the partialAccept.

Repro steps:
1. Trigger inline completion
2. get Suggest Dropdown
3. Trigger accept next word (ctrl+right)
4. No PartialAccept notification sent

Expected:
PartialAccept message would be sent for the current shown completion Item

This seems to occur as the edit application causes model update, which triggers disposal of inlineCompletions due to suggestItem being cleared.

VS Code version: Code 1.83.0 
OS version: ChromeOS Version 117.0.5938.144 (Official Build) (64-bit)
